### PR TITLE
executor: refactor analyze executor

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/35200

Problem Summary:

### What is changed and how it works?

separete the code in `analyze` executor into the following category:
1. column
2. index
3. incremental analyze
4. global stats
5. util
6. fast analyze

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by XuHuaiyu — APPROVED** [review]

File: executor/analyze.go [hunk 1]
```diff
@@ -116,148 +103,40 @@ func (e *AnalyzeExec) Next(ctx context.Context, req *chunk.Chunk) error {
...
-				logutil.BgLogger().Error("record historical stats failed", zap.Error(err))
-			}
-		}
-	}
 	for _, task := range e.tasks {
 		if task.colExec != nil && task.colExec.memTracker != nil {
 			task.colExec.memTracker.Detach()
 		}
 	}
+	pruneMode := variable.PartitionPruneMode(e.ctx.GetSessionVars().PartitionPruneMode.Load())
```
Comment (xuyifangreeneyes) [inline executor/analyze.go:111]: Maybe we need to switch the order between `memTracker.Detach` and `handleResultsError`. We wait for results coming from `resultsCh` in `handleResultsError` and before that, some of  `e.tasks` may be still running.

File: executor/analyze_global_stats.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+// globalStatsMap is a map used to store which partition tables and the corresponding indexes need global-level stats.
+// The meaning of key in map is the structure that used to store the tableID and indexID.
+// The meaning of value in map is some additional information needed to build global-level stats.
+type globalStatsMap map[globalStatsKey]globalStatsInfo
+
+func (e *AnalyzeExec) handleGlobalStats(ctx context.Context, needGlobalStats bool, globalStatsMap globalStatsMap) error {
+	statsHandle := domain.GetDomain(e.ctx).StatsHandle()
+	if needGlobalStats {
```
Comment (xuyifangreeneyes) [inline executor/analyze_global_stats.go:47]: Maybe we can early return when `needGlobalStats` is false.

Comment (Yisaer) [reply executor/analyze_global_stats.go:47]: In factor, it indeed return early when `needGlobalStats` is false

File: executor/analyze.go (hunk 1 above)
Comment (Yisaer) [reply executor/analyze.go:106]: updated

File: executor/analyze_global_stats.go (hunk 2 above)
Comment (xuyifangreeneyes) [reply executor/analyze_global_stats.go:47]: I mean early return in code style.
```
if !needGlobalStats {
    return nil
}
// do things
```
The current is also ok to me.

Comment (Yisaer) [reply executor/analyze_global_stats.go]: updated

File: executor/analyze.go [hunk 3]
```diff
@@ -116,76 +103,13 @@ func (e *AnalyzeExec) Next(ctx context.Context, req *chunk.Chunk) error {
...
-		} else {
-			finishJobWithLog(e.ctx, results.Job, nil)
-			// Dump stats to historical storage.
-			if err := e.recordHistoricalStats(results.TableID.TableID); err != nil {
-				logutil.BgLogger().Error("record historical stats failed", zap.Error(err))
-			}
-		}
+	err = e.handleResultsError(ctx, concurrency, needGlobalStats, globalStatsMap, resultsCh)
+	if err != nil {
+		return err
```
[the remark this answers is not in the document]
Comment (Yisaer) [reply executor/analyze.go]: updated, I move the detach code in front of `return error` which is the same behavior as before.

**Pull request merged**